src/include/daos_srv/daos_server.h

```diff
@@ -256,7 +294,7 @@ int dss_ult_create(void (*func)(void *), void *arg, int ult_type, int tgt_id,
 		   size_t stack_size, ABT_thread *ult);
 int dss_rebuild_ult_create(void (*func)(void *), void *arg, int ult_type,
 			   int tgt_id, size_t stack_size, ABT_thread *ult);
-int dss_ult_create_all(void (*func)(void *), void *arg);
+int dss_ult_create_all(void (*func)(void *), void *arg, bool main);
```

Comment:
there is no usage for the helper dss_ult_create_all() before, maybe can just remove it.
the dss_ult_create_all() usage in this patch seems can just use dss_thread_collective().

Reply:
The dss_thread_collective() will make the sponsor ULT busy-wait for the children ULTs to execute specified actions and collective the results. But the current requirement for DTX is to create ULT on each main XS, the sponsors ULT will NOT wait for the children ULTs that will run by themselves until the specified task is done (for current DTX case, it will run forever until the XS down).

Reply:
i see. BTW, I think it is not busy-wait.
keep using dss_ult_create_all() seems fine, or may define a NON_BLOCK flag to dss_collective() to ask it need not wait for completion. keep both dss_ult_create_all() and dss_collective looks a little confuse and duplicated, not a big deal, can do it later in other patch.